src/lib/TranscriptEditor/MediaPlayer/index.js

```diff
@@ -312,45 +305,16 @@ class MediaPlayer extends React.Component {
   }
 
   getMediaCurrentTime = () => {
-    if (this.videoRef.current !== null) {
-      return secondsToTimecode(this.videoRef.current.currentTime + this.state.timecodeOffset);
+    if (this.props.videoRef.current !== null) {
+      return secondsToTimecode(this.props.videoRef.current.currentTime + this.state.timecodeOffset);
     }
 
     return '00:00:00:00';
   }
-  handleMediaDurationChange =(e) => {
-    if (this.props.handleAnalyticsEvents !== undefined) {
-      this.props.handleAnalyticsEvents({
-        category: 'MediaPlayer',
-        action: 'mediaDuration',
-        name: secondsToTimecode(e.target.duration),
-        value: e.target.duration
-      });
-    }
-  }
-
-  onLoadedDataGetDuration = (e) => {
-    const currentDuration = e.target.duration;
-    const currentDurationWithOffset = currentDuration+ this.state.timecodeOffset;
-    const durationInSeconds = secondsToTimecode( currentDuration+ currentDurationWithOffset);
-
-    this.setState({
-      mediaDuration: durationInSeconds
-    });
-
-    if (this.props.handleAnalyticsEvents !== undefined) {
-      this.props.handleAnalyticsEvents({
-        category: 'MediaPlayer',
-        action: 'onLoadedDataGetDuration',
-        name: 'durationInSeconds-WithoutOffset',
-        value: secondsToTimecode( currentDuration)
-      });
-    }
-
-  }
 
   handlePictureInPicture = () => {
-    if (this.videoRef.current !== null) {
+    // console.log('this.props.videoRef', this.props.videoRef, this.props.videoRef.current );
+    if (this.props.videoRef.current !== undefined) {
```

Comment:
Excellent question, honest answer, I had figured it out with trial and error. 

Might be worth investigating.

here's what I found:

https://reactjs.org/docs/refs-and-the-dom.html

>React will assign the current property with the DOM element when the component mounts, and assign it back to null when it unmounts. ref updates happen before componentDidMount or componentDidUpdate lifecycle methods.

this explains the `null` but not the `undefined`(?)

On stackoverflow: 

https://stackoverflow.com/questions/35230020/this-refs-something-returns-undefined

>The correct place to work with refs is inside specific React lifecycle methods e.g. ComponentDidMount, ComponentDidUpdate
>
>You cannot reference refs from the render() method.

This could be linked with your previous point about a larger refactor (of `
this.props.videoRef.current !== null`)